mercury/views/can.py

```diff
@@ -29,12 +81,16 @@ def post(request, *args, **kwargs):
         return HttpResponse(e.error, status=400)
 
     try:
-        sensor_type, decoded_data = decoder.decode_can_message_full_dict()
-        return HttpResponse(json.dumps(decoded_data), status=201)
+        decoded_data = decoder.decode_can_message()
     except InvalidBitException as e:
         return HttpResponse(e.error, status=400)
-    except NotImplementedError as e:
-        return HttpResponse(e.args, status=400)
+    try:
+        sensor = create_and_save_sensor(decoded_data)
+    except KeyError as e:
+        return HttpResponse(f"Missing data for {e}", status=400)
+    log.debug("Saving a new instance of {}".format(sensor))
+    sensor.save()
+    return HttpResponse(json.dumps(decoded_data), status=201)
```

Comment:
Not sure how this is working. Can you please help me understand?

Reply:
Sure - but can you be more detailed about what you want to know?

`created_and_save_sensor()` takes in previously decoded data. If you look at that method, it figures out which Model the can_id maps to, and creates an instance of it with the data. The object returned is stored in variable `sensor` with the data mapped to the appropriate attributes of the model and then `save()` is called to write it to the DB.

Reply:
I see, thank you!